pkg/mvtxdaemon/server/service/rater/helper.go

```diff
@@ -79,6 +81,34 @@ func CalculateRate(q *sharedqueue.OverflowQueue[*TimestampedCounts], lookbackSec
 	return delta / float64(timeDiff)
 }
 
+// CalculatePending calculates the pending of a MonoVertex for a given lookback period.
+func CalculatePending(q *sharedqueue.OverflowQueue[*TimestampedCounts], lookbackSeconds int64) int64 {
+	counts := q.Items()
+	if len(counts) <= 1 {
+		return pendingNotAvailable
+	}
+	startIndex := findStartIndex(lookbackSeconds, counts)
+	// we consider the last but one element as the end index because the last element might be incomplete
+	// we can be sure that the last but one element in the queue is complete.
+	endIndex := len(counts) - 1
+	if startIndex == indexNotFound {
+		return pendingNotAvailable
+	}
+	delta := int64(0)
+	num := int64(0)
+	for i := startIndex; i <= endIndex; i++ {
+		currentPending := counts[i].PodCountSnapshot()
+		for _, pendingCount := range currentPending {
+			delta += int64(pendingCount)
+			num++
+		}
+	}
+	if num > 0 {
+		return delta / num
+	}
+	return delta
```

Comment:
```suggestion
        if num == 0 {
            return pendingNotAvailable
        }
        return delta / num
```

When num is zero we should return -1 right?

Reply:
Yes you are correct. Good catch.